azurerm/monitor_diagnostics.go

```diff
@@ -0,0 +1,48 @@
+package azurerm
+
+import "strings"
+
+type MonitorDiagnosticId struct {
+	ResourceID string
+	Name       string
+}
+
+func parseMonitorDiagnosticId(monitorId string) MonitorDiagnosticId {
+	var returnId MonitorDiagnosticId
+	returnId.ResourceID = retrieveResourceId(monitorId)
+	returnId.Name = retrieveDiagnosticName(monitorId)
+	return returnId
+}
+
+func retrieveResourceId(diagnosticSettingId string) string {
+	substring := "/providers/microsoft.insights/diagnosticSettings/"
+	return diagnosticSettingId[0:strings.Index(diagnosticSettingId, substring)]
+}
+
+func retrieveDiagnosticName(diagnosticSettingId string) string {
+	substring := "/providers/microsoft.insights/diagnosticSettings/"
+	return diagnosticSettingId[len(substring)+strings.Index(diagnosticSettingId, substring):]
+}
```

Comment:
on reflection I feel like a less brittle way of doing this would be to compose our own resource ID of `{targetResourceID}|{name}` since some of the Azure API's are case sensitive - I'm going to push a commit to fix this.